plugins/CoreConsole/Commands/DevelopmentSyncUITestScreenshots.php

```
@@ -73,42 +71,166 @@ protected function execute(InputInterface $input, OutputInterface $output)
+        return $this->getDiffviewContentForPrivatePlugin($input);
+    }
+
+    protected function getDiffviewContentForPrivatePlugin(InputInterface $input)
+    {
+        $httpUser = $input->getOption('http-user');
+        $httpPassword = $input->getOption('http-password');
+        if (empty($httpUser) || empty($httpPassword)) {
+            Log::info("--http-user and --http-password was not specified, skip download of private plugins screenshots.");
+            return;
+        }
+
+        // Attempt to download from protected/ artifacts...
+        $this->urlBase = str_replace("builds-artifacts.piwik.org/", "builds-artifacts.piwik.org/protected/", $this->urlBase);
+        $diffviewerUrl = $this->getDiffviewerUrl($this->urlBase);
+
+        return $this->downloadDiffviewer($diffviewerUrl, $httpUser, $httpPassword);
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDiffviewerUrl($urlBase)
+    {
+        return $urlBase . "/screenshot-diffs/diffviewer.html";
+    }
+
+    protected function downloadDiffviewer($urlDiffviewer, $httpUsername = false, $httpPassword = false)
+    {
+        $responseExtended = Http::sendHttpRequest(
+            $urlDiffviewer,
+            $timeout = 60,
+            $userAgent = null,
+            $destinationPath = null,
+            $followDepth = 0,
+            $acceptLanguage = false,
+            $byteRange = false,
+            $getExtendedInfo = true,
+            $httpMethod = 'GET',
+            $httpUsername,
+            $httpPassword
+        );
+        $httpStatus = $responseExtended['status'];
+        if ($httpStatus == '200') {
+            $diffviewer = str_replace('&', '&amp;', $responseExtended['data']);
+            return $diffviewer;
+        }
+
+        Log::info("Could not download the diffviewer from $urlDiffviewer - Got HTTP status " . $httpStatus);
+        if($httpStatus == '401') {
+            Log::warning("HTTP AUTH username and password are not valid.");
```

If this means the command fails, you might want to throw an exception here (possibly InvalidArgumentException) instead of logging a warning.